BUGPROCESS.md

```diff
@@ -0,0 +1,107 @@
+
+
+### 3. Explain or fix [confirmed issues](https://github.com/hashicorp/terraform/issues?q=is%3Aopen+label%3Abug+-label%3Aexplained+-label%3Abackend%2Foss+-label%3Abackend%2Fazure+-label%3Abackend%2Fs3+-label%3Abackend%2Fgcs+-label%3Abackend%2Fconsul+-label%3Abackend%2Fartifactory+-label%3Aterraform-cloud+-label%3Abackend%2Fremote+-label%3Abackend%2Fswift+-label%3Abackend%2Fpg+-label%3Abackend%2Ftencent++-label%3Abackend%2Fmanta++-label%3Abackend%2Fatlas++-label%3Abackend%2Fetcdv3++-label%3Abackend%2Fetcdv2+label%3Aconfirmed+-label%3A%22pending+project%22+)
+The next step for confirmed issues is to either:
+
+* explain why the behavior is expected, label the issue as "working as designed", and close it, or
+* locate the cause of the defect in the codebase. When the defect is located, and that description is posted on the issue, the issue is labeled "explained". In many cases, this step will get skipped if the fix is obvious, and engineers will jump forward and make a PR. 
+
+#### Goals
+Bugs with a high impact with widespread impact should be fixed in the next Z release.
+
+1. All [confirmed crashes](https://github.com/hashicorp/terraform/issues?q=is%3Aopen+label%3Acrash+label%3Abug+-label%3Aexplained+-label%3Abackend%2Foss+-label%3Abackend%2Fazure+-label%3Abackend%2Fs3+-label%3Abackend%2Fgcs+-label%3Abackend%2Fconsul+-label%3Abackend%2Fartifactory+-label%3Aterraform-cloud+-label%3Abackend%2Fremote+-label%3Abackend%2Fswift+-label%3Abackend%2Fpg+-label%3Abackend%2Ftencent++-label%3Abackend%2Fmanta++-label%3Abackend%2Fatlas++-label%3Abackend%2Fetcdv3++-label%3Abackend%2Fetcdv2+label%3Aconfirmed+-label%3A%22pending+project%22+) should be fixed in each Z release, even if they are unlikely to have a widespread impact
+2. [confirmed non-crash bugs](https://github.com/hashicorp/terraform/issues?q=is%3Aopen+-label%3Adocumentation+-label%3Acrash+label%3Abug+-label%3Aexplained+-label%3Abackend%2Foss+-label%3Abackend%2Fazure+-label%3Abackend%2Fs3+-label%3Abackend%2Fgcs+-label%3Abackend%2Fconsul+-label%3Abackend%2Fartifactory+-label%3Aterraform-cloud+-label%3Abackend%2Fremote+-label%3Abackend%2Fswift+-label%3Abackend%2Fpg+-label%3Abackend%2Ftencent+-label%3Abackend%2Fmanta+-label%3Abackend%2Fatlas+-label%3Abackend%2Fetcdv3+-label%3Abackend%2Fetcdv2+label%3Aconfirmed+-label%3A%22pending+project%22+sort%3Areactions-%2B1-desc) should be resolved roughly in order of upvotes. Most of these issues should be fixed within 4 weeks, and issues that are blocked on needing a larger research or refactor project should have a brief explanation posted and be labeled "pending project"
+
```

Comment:
I think there's something to be said here about bug recency and back porting, but I'm not sure what exactly to say. My understanding:

- Bugs related to recently-released work which have high impact should be fixed in a patch release where possible.
- Bugs which have existed for longer than the most recent major release are normally by definition less urgent, and do not normally warrant backporting to the stable branch and a patch release. Most commonly these will be released as part of the upcoming major release cycle. Exceptions may be made if the bug is somehow higher impact and a backport is feasible.